impl/src/main/java/org/eclipse/tractusx/wallet/stub/credential/impl/CredentialServiceImpl.java

```diff
@@ -63,11 +64,16 @@ public class CredentialServiceImpl implements CredentialService, InternalCredent
 
 
     @Override
-    public String getVerifiableCredentialByHolderBpnAndTypeAsJwt(String holderBpn, String type) {
+    public Pair<String, String> getVerifiableCredentialByHolderBpnAndTypeAsJwt(String holderBpn, String type) {
         try {
-            Optional<String> optionalVC = storage.getCredentialsAsJwtByHolderBpnAndType(holderBpn, type);
+            Optional<Pair<String, String>> optionalVC = storage.getCredentialsAsJwtByHolderBpnAndType(holderBpn, type);
             if (optionalVC.isPresent()) {
-                return optionalVC.get();
+                //if Id is URI , we need to extract the id part
+                String vcId = optionalVC.get().getLeft();
+                if(vcId.contains("#")){
```

Comment:
Are we 100% sure split result will always contain index 1? If not, Is that make sense for you to modify this structure little bit for possible incorrect URIs (e.g., "#" at the end) which could cause `ArrayIndexOutOfBoundsException`

Possible modification is in below
```suggestion
      if (vcId.contains("#")) {
      String[] parts = vcId.split("#", 2);
      if (parts.length == 2 && !parts[1].isBlank()) {
          vcId = parts[1];
      }
}
```

Reply:
Yes, This credential will be generated by wallet stub only.
Adding the check

Reply:
Addressed with https://github.com/Cofinity-X/entry-ssi-dim-wallet-stub/pull/3/commits/4d16061a851a054bfa21880a910aea4cc78e3062